src/main/java/sirius/biz/tycho/search/OpenSearchController.java

```diff
@@ -0,0 +1,233 @@
+ * a "comet" link which sends individual search results as genrated by each provider as single JSON objects, each
+ * in its own line (separated by a <tt>\n</tt>).
+ * <p>
+ * On the client side we use the <tt>oboe</tt> library which handles this kind of incoming JSON objects and renders
+ * any incoming results as fast as possible.
+ */
+@Register
+public class OpenSearchController extends BizController {
+
+    /**
+     * Contains a readily built byte array to separate messages.
+     * <p>
+     * The client library expects that we sent the results as "comets" - that are JSON objects separated by a new line.
+     * This way, we can immediatelly render the first results even while other providers are still searching for
+     * possible matches.
+     */
+    private static final byte[] NEWLINE = "\n".getBytes(StandardCharsets.UTF_8);
+
+    /**
+     * Specifies the maximal search duration before a search is canceled.
+     * <p>
+     * As we want to prevent a system overload if too many or too intense searches are performed, we abort once
+     * a query has been running for 5 seconds. Note that we don't necessarily interrupt any provider itself as
+     * this is not supported by the JVM - we however can close the connection to the client and free up the spot
+     * in the main thread pool.
+     */
+    private static final Duration SEARCH_TIMEOUT = Duration.ofSeconds(5);
+
+    /**
+     * To ensure that all possible results are rendered properly, we limit each provider to only 8 results at most.
+     */
+    private static final int MAX_RESULTS_PER_CATEGORY = 8;
+
+    /**
+     * Contains the name of the main thread pool which executes all searches.
+     */
+    private static final String TYCHO_OPEN_SEARCH_POOL = "tycho-open-search";
+
+    /**
+     * Contains the name of the inner thread pool which actually executes the query of a single provider. We use this
+     * additional thread pool so that a slow provider does not slow down other providers.
+     */
+    private static final String TYCHO_OPEN_SEARCH_TASK_POOL = "tycho-open-search-task";
+
+    private static final String PARAM_QUERY = "query";
+    private static final String RESPONSE_OVERLOAD = "overload";
+    private static final String CONTENT_TYPE_APPLICATION_JSON = "application/json";
+    private static final byte[] RESPONSE_COMPLETED_MESSAGE = "{\"completed\":true}\n".getBytes(StandardCharsets.UTF_8);
+    private static final String RESPONSE_CATEGORY = "category";
+    private static final String RESPONSE_CATEGORY_URL = "categoryUrl";
+    private static final String RESPONSE_PRIORITY = "priority";
+    private static final String RESPONSE_LABEL = "label";
+    private static final String RESPONSE_DESCRIPTION = "description";
+    private static final String RESPONSE_URL = "url";
+    private static final String RESPONSE_ACTIONS = "actions";
+
+    @Parts(OpenSearchProvider.class)
```

Comment:
Shouldnt this be priorized as well?

Reply:
nope, the results are - these providers are executed in parallel so sorting them doesn't make too much sense...